diff --git a/dlls/win32u/vulkan.c b/dlls/win32u/vulkan.c
index 49eac27..cca7759 100644
--- a/dlls/win32u/vulkan.c
+++ b/dlls/win32u/vulkan.c
@@ -24,6 +24,12 @@
 
 #include "config.h"
 
+/* SONAME_LIBVULKAN is used unconditionally, even with --without-vulkan.
+   Supply a library name for dlopen. */
+#ifndef SONAME_LIBVULKAN
+#define SONAME_LIBVULKAN "libvulkan.1.dylib"
+#endif
+
 #include <dlfcn.h>
 #include <pthread.h>
 #include <unistd.h>
